addon/services/segment.js

```diff
@@ -1,88 +1,148 @@
+    this.set('_defaultIdentifyUserDisabled', hasSegmentConfig && this.config.segment.defaultIdentifyUser === false)
+    this.set('_disabled', hasSegmentConfig && this.config.segment.enabled === false)
   },
 
-  hasAnalytics: function() {
+  hasAnalytics: function () {
     return !!(window.analytics && typeof window.analytics === "object");
   },
 
-  // Default true unless user explicitly sets defaultPageTrack to false
-  pageTrackEnabled: function() {
+  enabled: function () {
+    return !this.disabled();
+  },
+
+  disabled: function () {
+    return this.get('_disabled');
+  },
+
+  enable: function () {
+    this.set('_disabled', false);
+  },
+
+  disable: function () {
+    this.set('_disabled', true);
+  },
+
+  pageTrackEnabled: function () {
     return !this.pageTrackDisabled();
   },
 
-  pageTrackDisabled: function() {
-    const hasSegmentConfig = (this.config && this.config.segment);
-    return (hasSegmentConfig && this.config.segment.defaultPageTrack === false);
+  pageTrackDisabled: function () {
+    return this.get('_defaultPageTrackDisabled');
+  },
+
+  enableDefaultPageTrack: function () {
+    this.set('_defaultPageTrackDisabled', false);
   },
 
-  // Default true unless user explicitly sets defaultIdentifyUser to false
-  identifyUserEnabled: function() {
+  disableDefaultPageTrack: function () {
+    this.set('_defaultPageTrackDisabled', true);
+  },
+
+  identifyUserEnabled: function () {
     return !this.identifyUserDisabled();
   },
 
-  identifyUserDisabled: function() {
-    const hasSegmentConfig = (this.config && this.config.segment);
-    return (hasSegmentConfig && this.config.segment.defaultIdentifyUser === false);
+  identifyUserDisabled: function () {
+    return this.get('_defaultIdentifyUserDisabled');
+  },
+
+  enableDefaultIdentifyUser: function () {
+    this.set('_defaultIdentifyUserDisabled', false);
+  },
+
+  disableDefaultIdentifyUser: function () {
+    this.set('_defaultIdentifyUserDisabled', true);
   },
 
-  log: function() {
-    if(this.config && this.config.segment && this.config.segment.LOG_EVENT_TRACKING) {
-      Ember.Logger.info('[Segment.io] ', arguments);
+  log: function () {
+    if (this.config && this.config.segment && this.config.segment.LOG_EVENT_TRACKING) {
+      console.info('[Segment.io] ', arguments);// eslint-disable-line no-console
     }
   },
 
-  trackPageView: function() {
-    if(this.hasAnalytics()) {
+  trackPageView: function () {
+    if (this.enabled() && this.hasAnalytics()) {
       window.analytics.page.apply(this, arguments);
+      this.set('_calledPageTrack', true);
 
       this.log('trackPageView', arguments);
     }
   },
 
-  trackEvent: function(event, properties, options, callback) {
-    if(this.hasAnalytics()) {
+  trackEvent: function (event, properties, options, callback) {
+    if (this.enabled() && this.hasAnalytics()) {
+      this.checkPageTrackCalled();
```

Comment:
I believe this method call should not be in `trackEvent`.

Reply:
What method, `checkPageTrackCalled` ? It just logs a warning in console if `trackPageView` was never called. According to segment's documentation, some destinations require `page` call, so I added this.